docs/network-helpers/reference.md

```diff
@@ -0,0 +1,268 @@
+
+Example:
+
+```ts
+await helpers.setBalance(address, 100n**18n);
+```
+
+### `setCode(address, code)`
+
+Modifies the bytecode stored at an account's address.
+
+Parameters:
+
+- `address`: The address where the given code should be stored.
+- `code`: The code to store.
+
+Example:
+
+```ts
+await helpers.setCode(address, "0x1234...");
+```
+
+### `setNonce(address, nonce)`
+
+Modifies an account's nonce by overwriting it.
+
+Parameters:
+
+- `address`: The address whose nonce is to be changed.
+- `nonce`: The new nonce.
+
+Example:
+
+```ts
+await helpers.setNonce(address, 100);
+```
+
+### `setStorageAt(address, index, value)`
+
+Writes a single position of an account's storage.
+
+Parameters:
+
+- `address`: The address where the code should be stored.
+- `index`: The index in storage.
+- `value`: The value to store.
+
+Example:
+
+```ts
+await helpers.setStorageAt(address, storageSlot, newValue);
+```
+
+### `getStorageAt(address, index, [block])`
+
+Retrieves the data located at the given address, index, and block number.
+
+Parameters:
+
+- `address`: The address to retrieve storage from.
+- `index`: The position in storage.
+- `block`: The block number, or one of `"latest"`, `"earliest"`, or `"pending"`. Defaults to `"latest"`.
+
+Returns: A string containing the hexadecimal value retrieved.
+
+Example:
+
+```ts
+await helpers.getStorageAt(address, storageSlot);
+```
+
+### `impersonateAccount(address)`
+
+Allows Hardhat Network to sign transactions as the given address.
+
+Parameters:
+
+- `address`: The address to impersonate.
+
+Example:
+
+```ts
+await helpers.impersonateAccount(address);
+```
+
+### `stopImpersonatingAccount(address)`
+
+Stops Hardhat Network from impersonating the given address.
+
+Parameters:
+
+- `address`: The address to stop impersonating.
+
+Example:
+
+```ts
+await helpers.stopImpersonatingAccount(address);
+```
+
+## Time helpers
+
+### `latest()`
+
+Returns the timestamp of the latest block.
+
+Example:
+
+```ts
+await helpers.time.latest();
+```
+
+### `latestBlock()`
+
+Returns the number of the latest block.
+
+Example:
+
+```ts
+await helpers.time.latestBlock();
+```
+
+### `increase(amountInSeconds)`
+
+Mines a new block whose timestamp is `amountInSeconds` after the latest block's timestamp.
+
+Parameters:
+
+- `amountInSeconds`: Number of seconds to increase the next block's timestamp by.
+
+Returns: The timestamp of the mined block.
+
+Example:
+
+```ts
+// advance time by one hour and mine a new block
+await helpers.time.increase(3600);
+```
+
+### `increaseTo(timestamp)`
+
+Mines a new block whose timestamp is `timestamp`.
+
+Parameters:
+
+- `timestamp`: Must be bigger than the latest block's timestamp.
+
+Example:
+
+```ts
+await helpers.time.increaseTo(newTimestamp);
+```
+
+### `setNextBlockTimestamp(timestamp)`
+
+Sets the timestamp of the next block but doesn't mine one.
+
+Parameters:
+
+- `timestamp`: Can be `Date` or Epoch seconds. Must be greater than the latest block's timestamp.
```

Comment:
Might be good to add a link to some authoritative source on what "Epoch seconds" are.

Also, do the other helpers (which accept timestamps) also all accept `Date`? If so, would be good to mention it in the context of those other helpers.

Reply:
Added a link to wikipedia, and opened [an issue in Linear](https://linear.app/nomic-foundation/issue/HH-731/accept-date-instances-in-all-the-network-helpers-that-accept) for the `Date` thing.